Character.cpp

```diff
@@ -98,34 +105,16 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        if (hitPoints < *initialHitPoints)
-        {
-            hitPoints = *initialHitPoints;
-        }        
-        
-        hitPoints *= 1.1;
-        *initialHitPoints = hitPoints;
-        
-        if (attackDamage < *initialAttackDamage)
-        {
-            attackDamage = *initialAttackDamage;
-        }
-        
-        attackDamage *= 1.1;
-        *initialAttackDamage = attackDamage;
-
-        if (armor < *initialArmorLevel)
-        {
-            armor = *initialArmorLevel;
-        }
-        armor *= 1.1;
-        *initialArmorLevel = armor;
-        
+        restoreAndBoostStats (hitPoints, initialHitPoints.get());
+        restoreAndBoostStats (attackDamage, initialAttackDamage.get());
+        restoreAndBoostStats (armor, initialArmorLevel.get());
```

Comment:
There is no need to `get()` when you can dereference.
See:
https://en.cppreference.com/w/cpp/memory/unique_ptr/operator*
- [ ] revise